### DISCUSSION TITLE
use PAT instead of {{ github.token }} for the actual pulling of the github action, as workaround for rate-limit of 1000.

### DISCUSSION THREAD

**Discussion by maartenvanderhoef** [community/community#62056] [Enterprise]
### Select Topic Area

General

### Body

I work with Github Enterprise which I do not administrate. The limits are set to 1000 actions per hour. When doing many Github jobs we are hitting rate limiting on the resolving of certain Github actions. We do however also have another Service Token, which is allowed to do 5000 actions.

The error we receive is: `Error: API rate limit exceeded while resolving action xxxxx/yyyyy.`

Is it possible to somehow use the PAT token instead of the {{ github.token }} to actually retrieve ( & resolve ) the Github Action ?

**Comment by nirav-gajera** [comment]
Yes, it is possible to use a PAT token instead of the `{{ github.token }}` to retrieve and resolve a GitHub Action. To do this, you can use the following steps:

1. In your GitHub Actions workflow, create a variable called `PAT_TOKEN` and set it to the value of your PAT token.
2. In the step where you want to retrieve the GitHub Action, use the following syntax:

```
uses: xxxxx/yyyyy@x.x.x with:
  token: ${{ env.PAT_TOKEN }}
```

For example, if you have a PAT token called `my-pat-token` and you want to retrieve the `actions/setup-go` action, you would use the following syntax in your workflow:

```
uses: actions/setup-go@v1.12.5 with:
  token: ${{ env.PAT_TOKEN }}
```

**Reply by maartenvanderhoef** [reply]
Hi @nirav-gajera !

Are you very sure ? 

For some actions I received: `Unexpected input(s) 'token', valid inputs are ['name', 'path', 'if-no-files-found', 'retention-days']`

So not all actions have this and knowing this, I really doubt it is used for the actual resolving. Adding to that I still see the rate limit of github.token going up just as much.

**Comment by H777K** [comment]
Hi @maartenvanderhoef 

I have exactly the same issue, were you able to find a solution for this?

**Comment by qoomon** [comment]
Maybe my project could help, although I don't understand your situation. https://github.com/qoomon/actions--access-token
